Index: code/server/sv_client.c
===================================================================
--- code/server/sv_client.c	(revision 582)
+++ code/server/sv_client.c	(working copy)
@@ -1437,15 +1437,25 @@
 	// normal to spam a lot of commands when downloading
 	if ( !com_cl_running->integer && 
 		cl->state >= CS_ACTIVE &&
-		sv_floodProtect->integer && 
-		svs.time < cl->nextReliableTime ) {
-		// ignore any other text messages from this client but let them keep playing
-		// TTimo - moved the ignored verbose to the actual processing in SV_ExecuteClientCommand, only printing if the core doesn't intercept
-		clientOk = qfalse;
-	} 
+		sv_floodProtect->integer ) {
 
-	// don't allow another command for one second
-	cl->nextReliableTime = svs.time + 1000;
+		if ((unsigned) (svs.time - cl->lastReliableTime) < 1500u) {
+			// Allow two client commands every 1.5 seconds or so.
+			if ((cl->lastReliableTime & 1u) == 0u) {
+				cl->lastReliableTime |= 1u;
+			}
+			else {
+				// This is now at least our second client command in
+				// a period of 1.5 seconds.  Ignore it.
+				// TTimo - moved the ignored verbose to the actual processing in
+				// SV_ExecuteClientCommand, only printing if the core doesn't intercept
+				clientOk = qfalse;
+			}
+		}
+		else {
+			cl->lastReliableTime = (svs.time & (~1)); // Lowest bit 0.
+		}
+	}
 
 	SV_ExecuteClientCommand( cl, s, clientOk );
 
Index: code/server/server.h
===================================================================
--- code/server/server.h	(revision 582)
+++ code/server/server.h	(working copy)
@@ -153,7 +153,7 @@
 	int				downloadSendTime;	// time we last got an ack from the client
 
 	int				deltaMessage;		// frame last client usercmd message
-	int				nextReliableTime;	// svs.time when another reliable command will be allowed
+	int				lastReliableTime;
 	int				nextReliableUserTime; // svs.time when another userinfo change will be allowed
 	int				lastPacketTime;		// svs.time when packet was last received
 	int				lastConnectTime;	// svs.time when connection started
